client/src/components/pages/CreatePostPage/PublicServices/index.js

```diff
@@ -0,0 +1,195 @@
+          name="description"
+          validationMsg="Please input your description!"
+          placeholder="Enter Event Description"
+          min={10}
+          max={false}
+        />
+        <Form.Item
+          label={
+            <span>
+              Image&nbsp;
+              <Tooltip title="Image for Public Services">
+                <Icon type="info-circle" />
+              </Tooltip>
+            </span>
+          }
+        >
+          {getFieldDecorator("image", {
+            rules: [
+              {
+                required: true,
+                message: "Please input your image!",
+                whitespace: true
+              }
+            ]
+          })(
+            <Upload
+              style={{ width: "100%" }}
+              name="eventImage"
+              action="/upload.do"
+              listType="picture"
+            >
+              <Button size="large">
+                <Icon type="upload" /> Click to upload
+              </Button>
+            </Upload>
+          )}
+        </Form.Item>
+        <InputAntd
+          withTip={false}
+          label="Alt-Text"
+          tipInfo=""
+          getFieldDecorator={getFieldDecorator}
+          name="altText"
+          validationMsg="Please input Alt Text For Image!"
+          placeholder="Your Alt Text For Image"
+        />
+        <Divider style={{ margin: "20px 0" }} />
+        <InputGroup size="large">
+          <InputAntd
+            withTip={false}
+            label="Focus Keyword"
+            getFieldDecorator={getFieldDecorator}
+            name="focusKeyword"
+            validationMsg="Please input your keyword!"
+            placeholder="Your main keyword"
+          />
+        </InputGroup>
+        <Card
+          title={
+            <>
+              Event Title
+              <br />
+              <span style={{ color: "#277839" }}>
+                www.socialstreets.co/events/{urlType && urlType}
+              </span>
+            </>
+          }
+          bordered
+          style={{ width: "100%", marginBottom: "20px" }}
+        >
+          <TextAreaAntd
+            withTip={false}
+            style={{ fontSize: "15px" }}
+            label="Meta Description"
+            getFieldDecorator={getFieldDecorator}
+            name="metaDescription"
+            validationMsg="Please input your Meta Description!"
+            placeholder="Your main Meta Description"
+            min={5}
+            max={false}
+          />
+        </Card>
+        <Form.Item>
+          <Btn onClick={() => {}} type="primary" htmlType="submit">
+            Publish
+          </Btn>
+          <Btn
+            className="main--form-btn-gradient main--form-btn"
+            onClick={() => {}}
+            type="primary"
+            htmlType="submit"
+          >
+            Preview
+          </Btn>
+          <Btn
+            className="main--form-btn-black main--form-btn"
+            onClick={() => <Redirect to="/home" />}
```

Comment:
@aminalakhsham I don't think this style of Redirect will work, because it's a `<Redirect>` component which will only actually do a redirect if it is rendered or mounted on the page somewhere.
Here, it's just being returned by a click handler, which won't cause it to be put on the page anywhere.  Instead, you might want to redirect by calling `this.props.history.push('/home')` (you'll need a `this.props.history`, which you could get by wrapping this component in a `withRouter` HOC:
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/withRouter.md

Also I'd recommend moving that into a class function in this PublicServicesForm class, so you can avoid having an arrow function inside render (which is inefficient because it recreates that function in memory every time this component renders).  Then you can just write `onClick={this.onClickBack}` or something like that, here.

Reply:
good I'll put that in mind and do it in refactor time